From 4b19b37b82f52215190203f39559330f7fe0e989 Mon Sep 17 00:00:00 2001
From: sai surya teja <sssteja@habana.ai>
Date: Wed, 19 Apr 2023 04:26:32 +0300
Subject: [PATCH] [SW-140835] fwupd: rename reset_state functions

This patch aligns erom with preboot based on the patch below:
https://gerrit.habana-labs.com/#/c/267302/

There are two reset state function, which their name needs adjustment,
in order to better match their operation.

Change-Id: I21540127eeb7f737ac4d48dbfafc19044b9b92e4
---

diff --git a/lib/preboot/module_fwupd.c b/lib/preboot/module_fwupd.c
index 541b692..3a932f5 100644
--- a/lib/preboot/module_fwupd.c
+++ b/lib/preboot/module_fwupd.c
@@ -419,7 +419,7 @@
 	state->wfc_cmd = state->cmd;
 }
 
-static void module_fwupd_reset_state(struct fwupd_state *state)
+static void module_fwupd_internal_reset_state(struct fwupd_state *state)
 {
 	state->cmd.val = 0;	/* NOOP */
 	state->wfc_cmd.val = 0;	/* NOOP */
@@ -431,9 +431,9 @@
 	state->noret_delay = 0;
 }
 
-static void module_fwupd_reset_state_secure(struct fwupd_state *state)
+static void module_fwupd_reset_state(struct fwupd_state *state)
 {
-	module_fwupd_reset_state(state);
+	module_fwupd_internal_reset_state(state);
 	preboot_set_sram_sf_access(false);
 	preboot_security_dynamic_img_sram_close();
 	preboot_security_dynamic_img_elbi_reg_close();
@@ -498,7 +498,7 @@
 	case HL_UPD_CLR_STS:
 		break;
 	case HL_UPD_RST_STATE:
-		module_fwupd_reset_state_secure(state);
+		module_fwupd_reset_state(state);
 		module_fwupd_send_status(state, HL_UPD_STS_RST_OK, NULL);
 		break;
 	case HL_UPD_MAP_SRAM:
@@ -762,7 +762,7 @@
 	 *       by the host arrives.
 	 */
 	if (err) {
-		module_fwupd_reset_state(state);
+		module_fwupd_internal_reset_state(state);
 	} else if (state->exec_res == NOT_FOUND) {
 		state->wfc_cmd.val = 0;	/* NOOP - clear cmds state machine */
 		state->cmd.val = 0;	/* NOOP */
